src/Umbraco.Web/Compose/UnattendedInstallComponent.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Web;
+using Umbraco.Core;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Configuration;
+using Umbraco.Core.Exceptions;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Migrations.Install;
+using Umbraco.Core.Services;
+using Umbraco.Web.Install;
+using Umbraco.Web.Security;
+
+namespace Umbraco.Web.Compose
+{
+    public class UnattendedInstallComponent : IComponent
+    {
+        private readonly IRuntimeState _runtimeState;
+        private readonly ILogger _logger;
+        private readonly DatabaseBuilder _databaseBuilder;
+
+        public UnattendedInstallComponent(IRuntimeState runtimeState, ILogger logger, DatabaseBuilder databaseBuilder)
+        {
+            _runtimeState = runtimeState ?? throw new ArgumentNullException(nameof(runtimeState));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _databaseBuilder = databaseBuilder ?? throw new ArgumentNullException(nameof(databaseBuilder));
+        }
+
+        public void Initialize()
+        {
+            // check if we are doing an unattended install
+            if (_runtimeState.Reason != RuntimeLevelReason.InstallEmptyDatabase) return;
+
+            _logger.Info<UnattendedInstallComponent>("Installing Umbraco.");
+            var result = _databaseBuilder.CreateSchemaAndData();
+            _logger.Info<UnattendedInstallComponent>("Umbraco Installed.");
+
+            if (result.Success == false)
+                throw new InstallException("An error occurred while running the unattended installation.\n" + result.Message);
+
+            // the install was successful, lets restart the application to get Umbraco to boot the correct state
+            UmbracoApplication.Restart();
```

Comment:
Not sure if this is the best way/place to do this, as I'm not sure what happens if we want to run another component after this one, will it not run because the application is restarting?

Reply:
Updated to set a marker and do the restart in the `UmbracoInjectedModule` instead. This allows for running additional components after this one.